src/handshakes/abstract-handshake.ts

```diff
@@ -55,26 +52,13 @@ export abstract class AbstractHandshake {
     return uint8ArrayEquals(emptyKey, k)
   }
 
-  protected incrementNonce (n: uint64): uint64 {
-    return n + 1
-  }
-
-  protected nonceToBytes (n: uint64): bytes {
-    // Even though we're treating the nonce as 8 bytes, RFC7539 specifies 12 bytes for a nonce.
-    const nonce = new Uint8Array(12)
-    new DataView(nonce.buffer, nonce.byteOffset, nonce.byteLength).setUint32(4, n, true)
-
-    return nonce
-  }
-
-  protected encrypt (k: bytes32, n: uint64, ad: Uint8Array, plaintext: Uint8Array): bytes {
-    if (n > MAX_NONCE) {
+  protected encrypt (cs: CipherState, ad: Uint8Array, plaintext: Uint8Array): bytes {
```

Comment:
why the interface change?

Reply:
will revert it back to pass in `k` and `n` 👍